src/main/resources/templates/repo/step-functions-template.json.vpt

```diff
@@ -0,0 +1,272 @@
+#if( $athenaQueryDescriptors )
+
+	## Since this is a velocity template and we need a $ for referencing the AWS functions below 
+	## we set a d variable that represents the dollar sign
+
+	#set($d = "$")
+	
+	## The role is used by the step functions in order to run athena and SQS and read from the glue catalog 
+	,"${stack}${instance}AthenaQueryStateMachineExecutionRole": {
+		"Type": "AWS::IAM::Role",
+		"Properties": {
+			"AssumeRolePolicyDocument": {
+				"Version": "2012-10-17",
+				"Statement": [
+					{
+						"Effect": "Allow",
+						"Principal": {
+							"Service": [
+								"states.amazonaws.com"
+							]
+						},
+						"Action": [
+							"sts:AssumeRole"
+						]
+					}
+				]
+			},
+			"Path": "/",
+			"Policies": [
+				{
+					"PolicyName" : "${stack}${instance}AthenaQueryStateMachineExecutionPolicyAthena",
+					"PolicyDocument": {
+						"Version": "2012-10-17",
+						"Statement": [
+					        {
+								"Effect": "Allow",
+								"Action": [
+									"athena:startQueryExecution",
+						            "athena:stopQueryExecution",
+						            "athena:getQueryExecution"
+								],
+								"Resource": [
+									{ "Fn::Sub": "arn:aws:athena:${d}{AWS::Region}:${d}{AWS::AccountId}:workgroup/primary" }
+								]
+							},
+							{
+						        "Effect": "Allow",
+						        "Action": [
+						            "glue:GetDatabase",
+						            "glue:GetDatabases",
+						            "glue:GetTable",
+						            "glue:GetTables",
+						            "glue:GetPartition",
+						            "glue:GetPartitions",
+						            "glue:BatchGetPartition"
+						        ],
+						        "Resource": [
+						            { "Fn::Sub": "arn:aws:glue:${d}{AWS::Region}:${d}{AWS::AccountId}:catalog" },
```

Comment:
Can we make region and accoutId available in the global context and then use velocity ${} to fill in?